### PR TITLE
Added a module that checks if user owns the resource he's trying to modify

### PR DESCRIPTION
**Pull request by andela-oakinniranye**
A user should not be able to delete or update a resource he doesn't own, therefore this

### CODE REVIEW

File: app/controllers/answers_controller.rb [hunk 1]
```diff
@@ -26,13 +28,16 @@ def update
...
 
   def destroy
-    @answer.destroy
-    head :no_content
+    if @answer.try(:destroy)
+      head :no_content
+    else
+      invalid_request(error_msg)
+    end
   end
```
Comment (andela-ooranagwa) [inline app/controllers/answers_controller.rb]: Nice. I guess this skipped our attention earlier.

**Comment by andela-ooranagwa** [comment]
This should be extended to the `questions` and `comments` *controllers*

**Comment by andela-ooranagwa** [comment]
Awesome work :+1:

File: app/controllers/concerns/ownership_concern.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+
+require 'forwardable'
```
Comment (andela-ooranagwa) [inline app/controllers/concerns/ownership_concern.rb]: Could share more insight on this and what it is used for?

Comment (andela-oakinniranye) [inline app/controllers/concerns/ownership_concern.rb]: The ruby `forwardable` module allows you to delegate certain methods to other objects in your application.
Say you want to create a Queue system which uses an Array to queue items, one way to loop through would be to 
```ruby
def each(*item, &block)
queue.each{ |x|
 yield x if block_given?
}
end
```
However with the forwardable module, you could just delegate that method call to the appropriate object and let it handle it that way. So for example you could `def_delegator @queue, :each, :first, :last, :max, :min` and other array methods you would like to delegate to.
In my case `def_delegator self, :resource_name` which allowed me to call the class_method `resource_name` inside an instance_method

**Comment by andela-oakinniranye** [comment]
Yes @andela-ooranagwa that's why I created it, so the other related controllers can use it too. Don't know how it would be useful to the `CommentsController`, since it's handling both the questions and the answers. It can be figured out though

**Comment by andela-aadepoju** [comment]
:+1:

**Comment by andela-aadepoju** [comment]
:+1:

File: app/controllers/answers_controller.rb [hunk 3]
```diff
@@ -26,13 +28,16 @@ def update
...
-      render json: @answer.errors, status: :unprocessable_entity
+      invalid_request(error_msg)
     end
   end
 
   def destroy
-    @answer.destroy
-    head :no_content
+    if @answer.try(:destroy)
+      head :no_content
```
Comment (andela-echigbo) [inline app/controllers/answers_controller.rb]: Is it not required to state the response `status` when the response has `no content` like this case and the `update` action?

**Comment by andela-oojewale** [comment]
:+1:

File: app/controllers/answers_controller.rb (hunk 3 above)
Comment (andela-oakinniranye) [inline app/controllers/answers_controller.rb]: the `head :no_content` returns a response of `status: 204`. So it would not be required here

**Pull request merged**